x-pack/filebeat/module/o365/audit/config/pipeline.js

```diff
@@ -0,0 +1,401 @@
+            25: 'MicrosoftTeams', // Events from Microsoft Teams.
+            28: 'ThreatIntelligence', // Phishing and malware events from Exchange Online Protection and Office 365 Advanced Threat Protection.
+            30: 'MicrosoftFlow', // Microsoft Power Automate (formerly called Microsoft Flow) events.
+            31: 'AeD', // Advanced eDiscovery events.
+            32: 'MicrosoftStream', // Microsoft Stream events.
+            33: 'ComplianceDLPSharePointClassification', // Events related to DLP classification in SharePoint.
+            35: 'Project', // Microsoft Project events.
+            36: 'SharePointListOperation', // SharePoint List events.
+            38: 'DataGovernance', // Events related to retention policies and retention labels in the Security & Compliance Center
+            40: 'SecurityComplianceAlerts', // Security and compliance alert signals.
+            41: 'ThreatIntelligenceUrl', // Safe links time-of-block and block override events from Office 365 Advanced Threat Protection.
+            42: 'SecurityComplianceInsights', // Events related to insights and reports in the Office 365 security and compliance center.
+            44: 'WorkplaceAnalytics', // Workplace Analytics events.
+            45: 'PowerAppsApp', // Power Apps events.
+            47: 'ThreatIntelligenceAtpContent', // Phishing and malware events for files in SharePoint, OneDrive for Business, and Microsoft Teams from Office 365 Advanced Threat Protection.
+            49: 'TeamsHealthcare', // Events related to the Patients application in Microsoft Teams for Healthcare.
+            52: 'DataInsightsRestApiAudit', // Data Insights REST API events.
+            54: 'SharePointListItemOperation', // SharePoint list item events.
+            55: 'SharePointContentTypeOperation', // SharePoint list content type events.
+            56: 'SharePointFieldOperation', // SharePoint list field events.
+            64: 'AirInvestigation', // Automated incident response (AIR) events.
+            66: 'MicrosoftForms', // Microsoft Forms events.
+        },
+    }));
+    builder.Add("mapEventOutcome", makeMapper({
+        from: 'o365audit.ResultStatus',
+        to: 'event.outcome',
+        lowercase: true,
+        default: 'unknown',
+        skip_missing: true,
+        mappings: {
+            'success': 'success', // This one is necessary to map Success
+            'succeeded': 'success',
+            'partiallysucceeded': 'success',
+            'true': 'success',
+            'failed': 'failure',
+            'false': 'failure',
+        },
+    }));
+    builder.Add("setEventKind", new processor.AddFields({
+        target: 'event',
+        fields: {
+            kind: 'event',
+        },
+    }));
+    builder.Add("setUserFieldsFromId", new processor.Dissect({
+        tokenizer: "%{name}@%{domain}",
+        field: "user.id",
+        target_prefix: "user",
+        'when.contains.user.id': '@',
```

Comment:
Should we also try to parse the other form, like `SHAREPOINT\system`, where we have the user.domain and user.name? I also see some have a form like `NT AUTHORITY\\SYSTEM (Microsoft.Exchange.ServiceHost)`, but I'm not sure what the `Microsoft.Exchange.ServiceHost` means.

Reply:
These are all the different "user IDs" that I've captured:

```
ASR@TESTSIEM2.ONMICROSOFT.COM
DlpAgent
NT AUTHORITY\SYSTEM (Microsoft.Exchange.ServiceHost)
S-1-5-18
SecurityComplianceAlerts
Service Account
ServicePrincipal_148657c9-73a1-4f49-8a6d-051984dc9d5e
ServicePrincipal_e2aa77b1-3ac1-4bce-9f7e-cd2440b8e5c2
Unknown
alice@testsiem.onmicrosoft.com
alice@testsiem2.onmicrosoft.com
app@sharepoint
asr@testsiem.onmicrosoft.com
bob@testsiem.onmicrosoft.com
fim_password_service@support.onmicrosoft.com
Application
36787265537
36085768193
```
Different services seem to have a wildly different concept of an user ID. "NT AUTHORITY" is used in all messages from the "ExchangeAdmin" schema, and only in those.  I think it's better to not set "user.id" for those messages. WDYT?